packages/core/test/util/renderer.test.ts

```diff
@@ -296,3 +295,30 @@ test('createDefaultValue', t => {
   t.deepEqual(createDefaultValue({ type: 'object' }), {});
   t.deepEqual(createDefaultValue({ type: 'something' }), {});
 });
+
+test(`mapStateToDispatchRendererProps should generate UI schema given ownProps schema`, t => {
+  const store = mockStore(createState(coreUISchema));
+  const schema = {
+    type: 'object',
+    properties: {
+      bar: {
+        type: 'number'
+      }
+    }
+  };
+
+  const props = mapStateToDispatchRendererProps(
+    store.getState(),
+    { schema }
+  );
+  t.deepEqual(props.uischema, generateDefaultUISchema(schema));
+});
+
+test(`mapStateToDispatchRendererProps should use registered UI schema given no ownProps`, t => {
```

Comment:
shouldn't we add the test case where a uischema is passed in using the ownProps?